csrc/fusion.cpp

```diff
@@ -37,9 +37,9 @@ FusionGuard::~FusionGuard() {
 }
 
 Fusion* FusionGuard::getCurFusion() {
-  return ACTIVE_FUSION;
+  return const_cast<Fusion*>(ACTIVE_FUSION);
```

Comment:
Why make `ACTIVE_FUSION` const if we need to const_cast it here?

Reply:
> let me know if you have a strong reason otherwise.

I tend to avoid `const_cast` when possible. Also I don't really see the point of keeping something const under the covers if the only way to access it is casting away the const.

Reply:
Isn't `getCurFusion` designed to be used as non-constant? Almost always, when we grab a fusion, we mean to modify it.

Reply:
> I tend to avoid const_cast when possible.

I don't think it's possible to avoid `const_cast`. `active_fusion_` being non-const avoids `const_cast` here but introduces two `const_cast`s elsewhere. My latest commit tried to do what you suggested. Do you prefer that? 

> Also I don't really see the point of keeping something const under the covers if the only way to access it is casting away the const.

That's a fair point.

Reply:
I think Jacob's point was to not making ACTIVE_FUSION const?

Reply:
> I think Jacob's point was to not making ACTIVE_FUSION const?

Yes, that's what I did in the most recent commit (and that introduces two `const_cast`s elsewhere).

Reply:
The current code still stores `ACTIVE_FUSION` as a const pointer. If there are no const pointers anywhere we don't need const_cast.